distributed/scheduler.py

```diff
@@ -3415,14 +3416,17 @@ def worker_objective(self, ts: TaskState, ws: WorkerState) -> tuple:
         """
         dts: TaskState
         nbytes: Py_ssize_t
-        comm_bytes: Py_ssize_t = 0
+        comm_bytes: double = 0
+        xfers: Py_ssize_t = 0
         for dts in ts._dependencies:
             if ws not in dts._who_has:
                 nbytes = dts.get_nbytes()
-                comm_bytes += nbytes
+                # amortize transfer cost over all waiters
+                comm_bytes += nbytes / len(dts._waiters)
```

Comment:
Can you add an in-code comment explaining how this division amortizes cost? I assume this is again a "local topology" argument related to the fan-out tasks (https://github.com/dask/distributed/pull/5325#discussion_r711392611) where we try to "ignore" tasks which will likely end up everywhere anyhow?

Reply:
Will do. It's related to that, but actually a simpler idea. Basically, if we transfer to this worker now, that opens up the potential for N other tasks to run on this worker without transferring the data. So you could look at as, rather than this task paying the whole cost up front and others getting the benefit for free, all the sibling tasks split the cost of the transferring evenly between them. (That's an analogy of course—once transferred, the other tasks don't actually pay anything!)